### PR TITLE
fix(launcher): launcher.launch() should pass 'timeout' option

### PR DESCRIPTION
**Pull request by ishmal**
Addresses https://github.com/GoogleChrome/puppeteer/issues/4782

### CODE REVIEW

**Comment by ishmal** [comment]
@googlebot I fixed it.

**Comment by bobpearson** [comment]
@googlebot I consent.

File: test/launcher.spec.js [hunk 1]
```diff
@@ -239,6 +240,23 @@ module.exports.addTests = function({testRunner, expect, defaultBrowserOptions, p
...
       });
+      it('should pass the timeout parameter to browser.waitForTarget', async() => {
+        const expectedTimeout = 100000;
+        const options = Object.assign({}, defaultBrowserOptions, {
+          timeout: expectedTimeout
+        });
+        const originalProto = Browser.prototype.waitForTarget;
+        let resTimeout = 0;
+        // function() because we want Browser's 'this'
+        Browser.prototype.waitForTarget = async function(predicate, options) {
```
Comment (kblok) [inline test/launcher.spec.js:251]: The pattern we use to test timeout is setting the timeout to 1 and expecting to fail. [See](https://github.com/puppeteer/puppeteer/blob/eddb23b52175a3dad8cec4a5e9b3c13e58f6f144/test/page.spec.js#L501)

**Review by jschfflr — CHANGES_REQUESTED** [review]
@ishmal Thanks for your contribution! Could you address @kblok's comment and resolve the merge conflicts? Happy to get this landed :)

**Comment by jschfflr** [comment]
Closing this in favor of https://github.com/puppeteer/puppeteer/pull/7596.

**Pull request closed without merging**